packages/kitchn/src/components/Scroller/index.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+
+import { withDecorator } from "../../hoc/withDecorator";
+import { KitchnComponent } from "../../types";
+
+type Props = {
+  /**
+   * The width of the scroller.
+   */
+  width?: string | number;
+
+  /**
+   * The height of the scroller.
+   */
+  height?: string | number;
+
+  /**
+   * The scroller's children.
+   */
+  children?: React.ReactNode;
+};
+
+export type ScrollerProps = KitchnComponent<Props>;
+
+const ScrollerComponent = styled(({ children, ...props }: ScrollerProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [showTopGradient, setShowTopGradient] = useState(false);
+  const [showBottomGradient, setShowBottomGradient] = useState(false);
+  const [showLeftGradient, setShowLeftGradient] = useState(false);
+  const [showRightGradient, setShowRightGradient] = useState(false);
+
+  useEffect(() => {
+    const container = containerRef.current;
+    if (container) {
+      const updateGradients = () => {
+        const hasVerticalScrollbar =
+          container.scrollHeight > container.clientHeight;
+        const hasHorizontalScrollbar =
+          container.scrollWidth > container.clientWidth;
+
+        setShowTopGradient(container.scrollTop > 0 && hasVerticalScrollbar);
+        setShowBottomGradient(
+          container.scrollHeight - container.scrollTop >
+            container.clientHeight && hasVerticalScrollbar,
+        );
+        setShowLeftGradient(container.scrollLeft > 0 && hasHorizontalScrollbar);
+        setShowRightGradient(
+          container.scrollWidth - container.scrollLeft >
+            container.clientWidth && hasHorizontalScrollbar,
+        );
+      };
+
+      container.addEventListener("scroll", updateGradients);
+
+      updateGradients();
+
+      return () => {
+        container.removeEventListener("scroll", updateGradients);
+      };
+    }
+
+    return undefined;
+  }, [children]);
+
+  return (
+    <Container {...props}>
+      <ScrollerContainer ref={containerRef}>{children}</ScrollerContainer>
+      {showTopGradient && <TopGradient />}
+      {showBottomGradient && <BottomGradient />}
+      {showLeftGradient && <LeftGradient />}
+      {showRightGradient && <RightGradient />}
+    </Container>
+  );
+})<ScrollerProps>`
+  position: relative;
+  overflow: hidden;
+  height: ${(props) =>
+    typeof props.height === "number" ? `${props.height}px` : props.height};
+  width: ${(props) =>
```

Comment:
we dont need to set the width and height since we can already do it with the decorator

Reply:
Okay I'll remove them from here